tests/virt/vmmigration_test.py

```diff
@@ -268,6 +269,35 @@ def test_post_copy_status(self):
         assert stats['status'] == vmstatus.PAUSED
 
 
+class TestMigrationTimeout:
+
+    def test_disk_timeout_needed(self):
+        with fake.VM(xmldevices=read_data('disk_devices.xml')) as testvm:
+            timeout = testvm._disk_preparation_timeout()
+            assert timeout == 10.0
+
+    def test_no_disks(self):
+        with fake.VM() as testvm:
+            timeout = testvm._disk_preparation_timeout()
+            assert timeout == 0.0
+
+    def test_overall_timeout(self):
+        with fake.VM(xmldevices=read_data('disk_devices.xml')) as testvm:
+            timeout = testvm._migration_destination_prepare_timeout()
+            assert timeout == 46.0
+
+    def test_max_timeout(self):
+        with MonkeyPatchScope([
+                (vdsm.virt.vm, 'config',
+                 make_config(
+                     [('vars', 'migration_listener_timeout', '30'),
+                      ('vars', 'max_migration_listener_timeout', '5')])),
+        ]):
```

Comment:
Since you use now pytest, it is better to use the monkeypatch
fixture.

   CONFIG = make_config(...)

   def test_max_timeout(monkeypatch):
       monkeypatch.setattr(vdsm.virt.vm, 'config, CONFIG)